modules/api-import-export/src/main/java/org/wso2/carbon/apimgt/importexport/APIService.java

```diff
@@ -260,4 +262,89 @@ public Response importAPI(@Multipart("file") InputStream uploadedInputStream, @Q
+        if (APIImportExportConstants.STATUS_FALSE.equalsIgnoreCase(defaultProviderStatus)) {
+            isProviderPreserved = false;
+        }
+
+        try {
+            Response authorizationResponse = AuthenticatorUtil.authorizeUser(httpHeaders);
+
+            if (Response.Status.OK.getStatusCode() != authorizationResponse.getStatus()) {
+                return authorizationResponse;
+            }
+
+            AuthenticationContext authenticationContext = (AuthenticationContext) authorizationResponse.getEntity();
+            String tenantDomain = MultitenantUtils.getTenantDomain(authenticationContext.getUsername());
+            String currentUser = authenticationContext.getDomainAwareUsername();
+
+            if (!tenantDomain.equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME)) {
+                currentUser = currentUser + "@" + tenantDomain;
+            }
+            APIImportUtil.initializeProvider(currentUser);
+
+            //Temporary directory is used to create the required folders
+            String currentDirectory = System.getProperty(APIImportExportConstants.TEMP_DIR);
+            String createdFolders = File.separator +
+                    RandomStringUtils.randomAlphanumeric(APIImportExportConstants.TEMP_FILENAME_LENGTH) +
+                    File.separator;
+            File importFolder = new File(currentDirectory + createdFolders);
+            boolean folderCreateStatus = importFolder.mkdirs();
+
+            //API import process starts only if the required folder is created successfully
+            if (folderCreateStatus) {
+                String uploadFileName = APIImportExportConstants.UPLOAD_FILE_NAME;
+                String absolutePath = currentDirectory + createdFolders;
+                APIImportUtil.transferFile(uploadedInputStream, uploadFileName, absolutePath);
+
+                String extractedFolderName;
+                try {
+                    extractedFolderName = APIImportUtil.extractArchive(
+                            new File(absolutePath + uploadFileName), absolutePath);
+                } catch (APIImportException e) {
+                    return Response.status(Status.BAD_REQUEST).entity(e.getMessage()).build();
+                }
+
+                if (tenantDomain != null &&
+                        !org.wso2.carbon.utils.multitenancy.MultitenantConstants.SUPER_TENANT_DOMAIN_NAME
+                                .equals(tenantDomain)) {
+                    PrivilegedCarbonContext.startTenantFlow();
+                    PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain, true);
+                    isTenantFlowStarted = true;
+                }
+
+                APIImportUtil.updateAPI(apiID, absolutePath + extractedFolderName, currentUser, isProviderPreserved);
+
+                importFolder.deleteOnExit();
+                return Response.status(Status.CREATED).entity("API updated successfully.\n").build();
```

Comment:
any reason to include newline character to the response?

Reply:
fixed, can you check?